src/reactviews/pages/AddFirewallRule/addFirewallRule.component.tsx

```diff
@@ -0,0 +1,270 @@
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogBody,
+    DialogContent,
+    DialogSurface,
+    DialogTitle,
+    Dropdown,
+    Field,
+    Input,
+    Label,
+    Link,
+    makeStyles,
+    MessageBar,
+    Option,
+    OptionOnSelectData,
+    Radio,
+    RadioGroup,
+    SelectionEvents,
+    Spinner,
+} from "@fluentui/react-components";
+
+import { locConstants as Loc } from "../../common/locConstants";
+import { addFirewallRuleReadMoreUrl } from "../ConnectionDialog/connectionConstants";
+import { useFormStyles } from "../../common/forms/form.component";
+import { FirewallRuleSpec } from "../../../sharedInterfaces/firewallRule";
+import { AddFirewallRuleState } from "../../../sharedInterfaces/addFirewallRule";
+import { ApiStatus } from "../../../sharedInterfaces/webview";
+
+enum IpSelectionMode {
+    SpecificIp = "specificIp",
+    IpRange = "ipRange",
+}
+
+const useStyles = makeStyles({
+    ipInputBox: {
+        marginLeft: "8px",
+        marginRight: "8px",
+        width: "120px",
+    },
+});
+
+export const AddFirewallRuleDialog = ({
+    state,
+    addFirewallRule,
+    closeDialog,
+    signIntoAzure,
+}: {
+    state: AddFirewallRuleState;
+    addFirewallRule: (firewallRuleSpec: FirewallRuleSpec) => void;
+    closeDialog: () => void;
+    signIntoAzure: () => void;
+}) => {
+    const styles = useStyles();
+    const formStyles = useFormStyles();
+
+    const [selectedTenantId, setSelectedTenantId] = useState<string>(
+        state.tenants.length > 0 ? state.tenants[0].id : "",
+    );
+
+    const [tenantDisplayText, setTenantDisplayText] = useState(
+        state.tenants.length > 0 ? formatTenant(state.tenants[0]) : "",
+    );
+
+    // Update selected tenant when state.tenants changes
+    useEffect(() => {
+        if (state.tenants.length > 0) {
+            setSelectedTenantId(state.tenants[0].id);
+            setTenantDisplayText(formatTenant(state.tenants[0]));
+        }
+    }, [state.tenants]);
+
+    useEffect(() => {
+        if (state.clientIp) {
+            setStartIp(replaceLastOctet(state.clientIp, 0));
+            setEndIp(replaceLastOctet(state.clientIp, 255));
+        }
+    }, [state.clientIp]);
+
+    const [ruleName, setRuleName] = useState("ClientIPAddress_" + formatDate(new Date()));
+
+    const [ipSelectionMode, setIpSelectionMode] = useState(IpSelectionMode.SpecificIp);
+
+    const [startIp, setStartIp] = useState(replaceLastOctet(state.clientIp, 0));
+    const [endIp, setEndIp] = useState(replaceLastOctet(state.clientIp, 255));
+
+    const onTenantOptionSelect = (_: SelectionEvents, data: OptionOnSelectData) => {
+        setSelectedTenantId(data.selectedOptions.length > 0 ? data.selectedOptions[0] : "");
+        setTenantDisplayText(data.optionText ?? "");
+    };
+
+    return (
+        <Dialog open={true /* standalone dialog always open*/}>
+            <DialogSurface>
+                <DialogBody>
+                    <DialogTitle>
```

Comment:
Will change this to look like a proper page later (rather than an empty page with a dialog over it in standalone mode).  All the logic tie-ins was the main fish to fry for now.